src/main/java/seedu/address/logic/parser/task/AddTaskCommandParser.java

```diff
@@ -20,11 +28,18 @@ public class AddTaskCommandParser implements Parser<AddTaskCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public AddTaskCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DEADLINE, PREFIX_CONTACT);
+
+        if (argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTaskCommand.MESSAGE_USAGE));
+        }
 
         Title title = TaskParserUtil.parseTitle(argMultimap.getPreamble());
+        Deadline deadline = TaskParserUtil.parseDeadline(argMultimap.getValue(PREFIX_DEADLINE).get());
```

Comment:
Should we also check if argMultimap.getValue(PREFIX_DEADLINE) is empty before getting the value? else it will throw an uncaught NoSuchElementException if the user forgets to enter a deadline/enter in the wrong format!

Reply:
Yup, there is a deeply nested function which does the checking, which I might refactor.